libethereum/Block.cpp

```diff
@@ -692,6 +693,29 @@ void Block::performIrregularModifications()
 	}
 }
 
+void Block::updateBlockhashContract()
+{
+	u256 const blockNumber = info().number();
+
+	u256 const metropolisForkBlock = m_sealEngine->chainParams().u256Param("metropolisForkBlock");
+	if (blockNumber == metropolisForkBlock)
+	{
+		m_state.createContract(c_blockhashContractAddress);
+		m_state.setNewCode(c_blockhashContractAddress, bytes(c_blockhashContractCode));
+		m_state.commit(State::CommitBehaviour::KeepEmptyAccounts);
+	}
+
+	if (blockNumber >= metropolisForkBlock)
+	{
+		u256 const nonce = transactionsFrom(SystemAddress);
+		u256 const gas = 1000000;
+		Transaction tx(0, 0, gas, c_blockhashContractAddress, m_previousBlock.hash().asBytes(), nonce);
```

Comment:
Again out crappy vector_ref interface, but I think you can go from h256 to bytes_ref without going through std::vector. It might look like `{hash.begin(), hash.size()}`. Or improve vector_ref.